endpoint.bal

```diff
@@ -0,0 +1,372 @@
+        return handleUploadPhotoResponse(uploadResponse);
+    }
+
+    # Delete a contact photo.
+    # 
+    # + resourceName - Contact resource name
+    # + return - () on success, else an 'error'
+    @display {label: "Delete a Contact Photo"}
+    isolated remote function deleteContactPhoto(@display {label: "Contact Resource Name"} string resourceName) returns 
+                                       @tainted error? {
+        string path = SLASH + resourceName + COLON + DELETE_PHOTO_PATH;
+        http:Response deleteResponse = <http:Response>check self.googleContactClient->delete(path);
+        return handleDeleteResponse(deleteResponse);
+    }
+
+    # Batch get contacts.
+    # 
+    # + resourceNames - String array of contact resource names
+    # + personFields - Restrict which fields on the person are returned
+    # + return - `PersonResponse[]` on success, else an `error`
+    @display {label: "Batch Get Contacts"}   
+    isolated remote function batchGetContacts(@display {label: "Contact Resource Names"} string[] resourceNames, 
+                                     @display {label: "Person Fields"} ContactMasks[] personFields) returns 
+                                     @tainted @display {label: "Array of PersonResponse"} PersonResponse[]|error {
+        string path = SLASH + BATCH_CONTACT_PATH;
+        string pathWithResources = prepareResourceString(path, resourceNames);
+        string pathWithPersonFields = prepareUrlWithPersonFields(pathWithResources + AMBERSAND, personFields);
+        http:Response httpResponse = <http:Response>check self.googleContactClient->get(pathWithPersonFields);
+        var response = check handleResponse(httpResponse);
+        var batchResponse = check response.cloneWithType(BatchGetResponse);
+        PersonResponse[] persons = [];
+        int i = persons.length();
+        foreach var result in batchResponse.responses {
+            var personResult = result.person;
+            if(personResult is json) {
+                persons[i] = check personResult.cloneWithType(PersonResponse);
+                i = i + 1;
+            }
+        }
+        return persons;
+    }
+
+    # Update a contact.
+    # 
+    # + resourceName - Contact resource name
+    # + updatePersonFields - Restrict which fields on the person are returned
+    # + personFields - Restrict which fields on the person are returned
+    # + return - `Person` on success else an `error`
+    @display {label: "Update a Contact"}  
+    isolated remote function updateContact(@display {label: "Contact Resource Name"} string resourceName, 
+                                  @display {label: "Contact details"} Person updateContact, 
+                                  @display {label: "Person Fields to be Updated"} ContactMasks[] updatePersonFields,
+                                  @display {label: "Person Fields"} ContactMasks[]? personFields = ()) returns 
```

Comment:
Update contact should be same as create contact. Additionally it will have contact resource name. As a user, it is no clear to me what should come in `updatePersonFields ` and what should come in `personFields `.

Reply:
Filter options should be specified